core/src/test/scala/caliban/parsing/ParserSpec.scala

```diff
@@ -492,6 +492,55 @@ object ParserSpec extends DefaultRunnableSpec {
           )
         )
       },
+      testM("input with no body") {
+        val inputWithNoBody = "input BarBaz"
+        assertM(Parser.parseQuery(inputWithNoBody))(
+          equalTo(
+            Document(
+              List(
+                InputObjectTypeDefinition(
+                  description = None,
+                  name = "BarBaz",
+                  directives = Nil,
+                  fields = Nil
+                )
+              ),
+              sourceMapper = SourceMapper.apply(inputWithNoBody)
+            )
+          )
+        )
+      },
+      testM("input with empty body") {
```

Comment:
These two tests yield different results. The first evaluates to:

```scala
Document(List(Input Object)
```


While the second evaluates to:

```scala
Document(List(Input Object, OperationDefinition(Query,None,List(),List(),List()))
```

i.e., the second one (with the `{ }` has an `OperationDefinition`, albeit an empty one). Is this expected?

Reply:
It's not expected. I think you need to switch the 2 parsers to catch the one with body first, then the one without. Otherwise it catches the one without body and the { } are left to be parsed. 

Also you need to change the parser for Scala 3 (Fastparse is not available for Scala 3 so cats-parse is used).

Reply:
Sounds good. I was looking to fix the Scala 2 Parser first to see if I'm doing it right. I'll push up the cats-parse fix once I finalize this one.